test/sharness/t0100-name.sh

```diff
@@ -8,209 +8,282 @@ test_description="Test ipfs repo operations"
+        '
+
+        test_expect_success "'ipfs name resolve' succeeds" '
+        ipfs name resolve "$PEERID" >output
+        '
+
+        test_expect_success "resolve output looks good" '
+        printf "/ipfs/%s/help\n" "$HASH_WELCOME_DOCS" >expected4 &&
+        test_cmp expected4 output
+        '
+
+        test_expect_success "ipfs cat on published content succeeds" '
+        ipfs cat "/ipfs/$HASH_WELCOME_DOCS/help" >expected &&
+        ipfs cat "/ipns/$PEERID" >actual &&
+        test_cmp expected actual
+        '
+
+        # publish with an explicit node ID
+
+        test_expect_failure "'ipfs name publish --allow-offline <local-id> <hash>' succeeds" '
+        echo ipfs name publish --allow-offline "${PEERID}" "/ipfs/$HASH_WELCOME_DOCS" &&
+        ipfs name publish --allow-offline "${PEERID}" "/ipfs/$HASH_WELCOME_DOCS" >actual_node_id_publish
+        '
+
+        test_expect_failure "publish with our explicit node ID looks good" '
+        echo "Published to ${PEERID}: /ipfs/$HASH_WELCOME_DOCS" >expected_node_id_publish &&
+        test_cmp expected_node_id_publish actual_node_id_publish
+        '
+
+        # test publishing with B36CID and B58MH resolve to the same B36CID
+
+        test_expect_success "generate and verify a new key" '
+        B58MH_ID=`ipfs key list -f=b58mh -l | grep self | cut -d " " -f1` &&
+        B36CID_ID=`ipfs key list -f=b36cid -l | grep self | cut -d " " -f1` &&
+        test_check_peerid "${B58MH_ID}" &&
+        test_check_peerid "${B36CID_ID}"
+        '
+
+        test_expect_success "'ipfs name publish --allow-offline --key=<peer-id> <hash>' succeeds" '
+        ipfs name publish --allow-offline  --key=${B58MH_ID} "/ipfs/$HASH_WELCOME_DOCS" >b58mh_published_id &&
+        ipfs name publish --allow-offline  --key=${B36CID_ID} "/ipfs/$HASH_WELCOME_DOCS" >b36cid_published_id
+        '
+
+        test_expect_success "publish an explicit node ID as two key in B58MH and B36CID, name looks good" '
+        echo "Published to ${B36CID_ID}: /ipfs/$HASH_WELCOME_DOCS" >expected_published_id &&
+        test_cmp expected_published_id b58mh_published_id &&
+        test_cmp expected_published_id b36cid_published_id
+        '
+
+        test_expect_success "'ipfs name resolve' succeeds" '
+        ipfs name resolve "$B36CID_ID" >output
+        '
+
+        test_expect_success "resolve output looks good" '
+        printf "/ipfs/%s\n" "$HASH_WELCOME_DOCS" >expected2 &&
+        test_cmp expected2 output
+        '
+
+        # test IPNS + IPLD
+
+        test_expect_success "'ipfs dag put' succeeds" '
+        HELLO_HASH="$(echo "\"hello world\"" | ipfs dag put)" &&
+        OBJECT_HASH="$(echo "{\"thing\": {\"/\": \"${HELLO_HASH}\" }}" | ipfs dag put)"
+        '
+        test_expect_success "'ipfs name publish --allow-offline /ipld/...' succeeds" '
+        PEERID=`ipfs key list -f=b36cid -l | grep self | cut -d " " -f1` &&
+        test_check_peerid "${PEERID}" &&
```

Comment:
Any reason we're redefining PEERID and checking it again, or just copy-paste?

Reply:
copy/paste.